From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Steinborn <git@steinborn.me>
Date: Sun, 8 Aug 2021 00:52:54 -0400
Subject: [PATCH] (Paper) Optimize entity tracker passenger checks


diff --git a/src/main/java/net/minecraft/server/level/EntityTrackerEntry.java b/src/main/java/net/minecraft/server/level/EntityTrackerEntry.java
index ffcf6e2dfc1519c6ff522999a7ffe1bce1fb47f2..9facdc8b41620c4ce9ec398286d2067de291e98d 100644
--- a/src/main/java/net/minecraft/server/level/EntityTrackerEntry.java
+++ b/src/main/java/net/minecraft/server/level/EntityTrackerEntry.java
@@ -88,7 +88,7 @@ public class EntityTrackerEntry {
         this.trackedPlayers = trackedPlayers;
         // CraftBukkit end
         this.m = Vec3D.ORIGIN;
-        this.p = Collections.emptyList();
+        this.p = com.google.common.collect.ImmutableList.of(); // Paper - optimize passenger checks
         this.b = worldserver;
         this.f = consumer;
         this.tracker = entity;
