From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Noah van der Aa <ndvdaa@gmail.com>
Date: Sat, 21 Aug 2021 16:32:07 +0200
Subject: [PATCH] (Paper) Respect despawn rate in merge check


diff --git a/src/main/java/net/minecraft/world/entity/item/EntityItem.java b/src/main/java/net/minecraft/world/entity/item/EntityItem.java
index 7513ca8ef416ee9ce87ba3faf0a2d2a1a58c82c1..db4b4da5e6420b49d0e9b81b03bee754f3f776d4 100644
--- a/src/main/java/net/minecraft/world/entity/item/EntityItem.java
+++ b/src/main/java/net/minecraft/world/entity/item/EntityItem.java
@@ -268,7 +268,7 @@ public class EntityItem extends Entity {
     private boolean z() {
         ItemStack itemstack = this.getItemStack();
 
-        return this.isAlive() && this.pickupDelay != 32767 && this.age != -32768 && this.age < 6000 && itemstack.getCount() < itemstack.getMaxStackSize();
+        return this.isAlive() && this.pickupDelay != 32767 && this.age != -32768 && this.age < this.getDespawnRate() && itemstack.getCount() < itemstack.getMaxStackSize(); // Paper - respect despawn rate in pickup check.
     }
 
     private void a(EntityItem entityitem) {
